domain/consensus/utils/consensushashing/calculate_signature_hash.go

```diff
@@ -14,16 +14,26 @@ type SigHashType uint32
 
 // Hash type bits from the end of a signature.
 const (
-	SigHashAll          SigHashType = 0x1
-	SigHashNone         SigHashType = 0x2
-	SigHashSingle       SigHashType = 0x3
-	SigHashAnyOneCanPay SigHashType = 0x80
+	SigHashAll          SigHashType = 0b00000001
+	SigHashNone         SigHashType = 0b00000010
+	SigHashSingle       SigHashType = 0b00000100
+	SigHashAnyOneCanPay SigHashType = 0b10000000
 
 	// SigHashMask defines the number of bits of the hash type which is used
 	// to identify which outputs are signed.
-	SigHashMask = 0x1f
+	SigHashMask = 0b00000111
 )
 
+func (sht SigHashType) isStandardSigHashType() bool {
+	switch sht {
+	case SigHashAll, SigHashNone, SigHashSingle,
+		SigHashAll | SigHashAnyOneCanPay, SigHashNone | SigHashAnyOneCanPay, SigHashSingle | SigHashAnyOneCanPay:
+		return true
+	default:
+		return false
+	}
+}
+
```

Comment:
Are we sure this is good and correct?

Reply:
As sure as one can be.
Many things inside the code and tests of CalculateSignatureHash assume the above is correct. 
If it is not - we should fix both it and the code.